components/brave_rewards/core/wallet_provider/connect_external_wallet.cc

```diff
@@ -8,23 +8,30 @@
 #include <utility>
 
 #include "base/containers/contains.h"
+#include "base/functional/bind.h"
+#include "base/location.h"
 #include "brave/components/brave_rewards/core/common/random_util.h"
 #include "brave/components/brave_rewards/core/database/database.h"
+#include "brave/components/brave_rewards/core/endpoints/request_for.h"
 #include "brave/components/brave_rewards/core/logging/event_log_keys.h"
 #include "brave/components/brave_rewards/core/logging/event_log_util.h"
 #include "brave/components/brave_rewards/core/rewards_engine_impl.h"
 #include "brave/components/brave_rewards/core/state/state_keys.h"
 #include "brave/components/brave_rewards/core/wallet/wallet_util.h"
 
 namespace brave_rewards::internal {
-
+using endpoints::GetWallet;
 using endpoints::PostConnect;
+using endpoints::RequestFor;
 using wallet::GetWalletIf;
 
 namespace wallet_provider {
 
 ConnectExternalWallet::ConnectExternalWallet(RewardsEngineImpl& engine)
-    : engine_(engine) {}
+    : engine_(engine) {
+  linkage_checker_.Start(FROM_HERE, base::Minutes(1), this,
```

Comment:
Does `this` need to be `base::Unretained(this)`?

Reply:
It's done internally:
```
template <class Receiver>
void Start(const Location& posted_from,
           TimeDelta delay,
           Receiver* receiver,
           void (Receiver::*method)()) {
  Start(posted_from, delay, BindRepeating(method, Unretained(receiver)));
}
```